CMakeLists.txt

```diff
@@ -213,6 +213,18 @@ if(OpenCL_FOUND)
   endif()
 endif()
 
+# search for amgcl
+find_file(AMG_HPP amgcl/amg.hpp HINTS ${AMGCL_INCLUDE_DIRS})
+if(AMG_HPP)
+  set(HAVE_AMGCL 1)
+  set(AMGCL_FOUND 1)
+  include_directories(${AMGCL_INCLUDE_DIRS})
+  set_source_files_properties(opm/simulators/linalg/bda/amgclSolverBackend.cpp PROPERTIES LANGUAGE CUDA)
+else()
+  set(HAVE_AMGCL 0)
+  set(AMGCL_FOUND 0)
+endif()
```

Comment:
It seems like amgcl install package configuration scripts [here](https://github.com/ddemidov/amgcl/blob/c781719e2f961c161a7ff0cf2c168ac6949ac510/CMakeLists.txt#L309-L311), shouldn't we just do
```
find_package(amgcl)
```
that should set `amgcl_FOUND`

Reply:
f0cd5cb